extensions/ql-vscode/src/sarif-utils.ts

```diff
@@ -57,68 +59,78 @@ export function parseSarifPlainTextMessage(message: string): SarifMessageCompone
  * @returns A string that is valid for the `.file` field of a `FivePartLocation`:
  * directory separators are normalized, but drive letters `C:` may appear.
  */
-export function getPathRelativeToSourceLocationPrefix(sourceLocationPrefix: string, sarifRelativeUui: string) {
+export function getPathRelativeToSourceLocationPrefix(
+  sourceLocationPrefix: string,
+  sarifRelativeUui: string
+) {
   const normalizedSourceLocationPrefix = sourceLocationPrefix.replace(/\\/g, '/');
-  return path.join(normalizedSourceLocationPrefix, decodeURIComponent(sarifRelativeUui));
+  return `file:${
+    path.join(normalizedSourceLocationPrefix, decodeURIComponent(sarifRelativeUui))
+    }`;
 }
 
-export function parseSarifLocation(loc: Sarif.Location, sourceLocationPrefix: string): ParsedSarifLocation {
+export function parseSarifLocation(
+  loc: Sarif.Location,
+  sourceLocationPrefix: string
+): ParsedSarifLocation {
   const physicalLocation = loc.physicalLocation;
   if (physicalLocation === undefined)
-    return { t: 'NoLocation', hint: 'no physical location' };
+    return { hint: 'no physical location' };
   if (physicalLocation.artifactLocation === undefined)
-    return { t: 'NoLocation', hint: 'no artifact location' };
+    return { hint: 'no artifact location' };
   if (physicalLocation.artifactLocation.uri === undefined)
-    return { t: 'NoLocation', hint: 'artifact location has no uri' };
+    return { hint: 'artifact location has no uri' };
 
   // This is not necessarily really an absolute uri; it could either be a
   // file uri or a relative uri.
   const uri = physicalLocation.artifactLocation.uri;
 
+  // FIXME: This is probably wrong
```

Comment:
What's going on here? Why are you changing it to return the full uri instead of the path?

Reply:
I should have removed that comment, since I have tested and it is working.

The old style used a path, but the new (sarif style) uses a uri. It's annoyingly complex, but at least we're only using one style here. I can work on this and make it easier to understand. Probably a unit test or two would be good as well.